.github/workflows/node-types.yml

```
@@ -0,0 +1,21 @@
+on:
+  workflow_call:
+
+jobs:
+  types-check:
+    name: "Check generated types are up-to-date"
+    if: ${{ !( github.event_name == 'pull_request' && github.event.action == 'closed' ) }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - run: npm ci
+      - run: npm run json:generate:types && npx prettier --write src/types/
```

```suggestion
      - run: npm run json:generate:types
```

I think this `prettier` call is now superfluous since it's done in the `json:generate:types`